components/sup/src/event.rs

```diff
@@ -123,106 +258,27 @@ impl EventStream {
     }
 }
 
-/// Defines the logic for transforming concrete event into a
-/// byte representation to publish to the event stream.
-// TODO: The ultimate format we need is not well defined yet, so we're
-// using the absolute simplest thing that could possibly "work". Maybe
-// it'll be JSON, maybe it'll be protobuf, maybe it'll be something
-// else. Whatever it ultimately becomes, this is where the
-// transformation logic will live.
-pub trait Event: Debug {
-    fn render(&self, core: &EventCore) -> Vec<u8> {
-        format!("{:?} - {:?}", core, self).into_bytes()
-    }
-}
-
 ////////////////////////////////////////////////////////////////////////
 
-/// All messages are published under this subject.
-const HABITAT_SUBJECT: &str = "habitat";
+/// How long should we for the event thread to start up before
+/// abandoning it and shutting down?
+#[derive(Clone, Debug)]
+struct EventThreadStartupWait(u64);
```

Comment:
In cases like this, I have an ever-so-slight preference for using the `u64`, since it tends to make all the other accompanying trait implementations a little bit simpler. I generally only need it to be a `Duration` at the final point of consumption, so turning it into a `Duration` directly in the `Into<Duration>` implementation seemed more straightforward all around.

(Were the use case of this type to become more complicated, then switching to a Duration internally could end up simplifying things.)

Happy to change to use a named `secs` field, though.